### PR TITLE
JBPM-5233: Data source management

### PR DESCRIPTION
**Pull request by wmedvede**
- Refactoring and tests added for DataSourceDefEditorService and DriverDefEditorService

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @manstis would you mind review this commit.
Thanks.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    @Override
+    protected String getOriginalDefString() {
+        return DataSourceDefSerializer.serialize( originalDataSourceDef );
+    }
+
+    @Override
+    protected void verifyCreateConditions( boolean global ) {
+        //we wants that:
+        try {
+            // 2) the definition was deployed
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Your numbering seems a bit wrong; should this be "1)" and (on L#130) "2)"?

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            expectedEvent = new NewDataSourceEvent( dataSourceDef, project, SESSION_ID, IDENTITY );
+        }
+        verify( newDataSourceEvent, times( 1 ) ).fire( expectedEvent );
+    }
+
+    @Override
+    protected void verifySaveConditions() {
+        //we wants that
+        try {
+            // 3) the definition was deployed
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Numbering sequence wrong?

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+    @Override
+    protected void verifyCreateConditions( boolean global ) {
+        //we wants that:
+        try {
+            // 2) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDataSource( dataSourceDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
+        }
+        // 3) the notification was fired.
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Numbering sequence wrong?

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+    @Override
+    protected void verifySaveConditions() {
+        //we wants that
+        try {
+            // 3) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDataSource( dataSourceDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
+        }
+        // 4) the update notification was fired.
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Numbering sequence wrong?

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        verify( updateDataSourceEvent, times( 1 ) ).fire( new UpdateDataSourceEvent( originalDataSourceDef, dataSourceDef, project, SESSION_ID, IDENTITY ) );
+    }
+
+    @Override
+    protected void verifyDeleteConditions() {
+        //we wants that
+        try {
+            // 2) the definition was un-deployed.
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Numbering sequence wrong?

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+            verify( runtimeManager, times( 1 ) ).unDeployDataSource( dataSourceDeploymentInfo, UnDeploymentOptions.forcedUnDeployment() );
+        } catch ( Exception e ) {
+        }
+        // 3) the delete notification was fired.
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Numbering sequence wrong?

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    @Override
+    protected void verifyCreateConditions( boolean global ) {
+        //we wants that:
+        try {
+            // 2) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDataSource( dataSourceDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
+        }
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Do you want to fail the test in the event of an exception? The ```runtimeManager.deployDayaSource(...)``` *could* be called - and hence pass the assertion, but generate an error that would be silently ignored.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    @Override
+    protected void verifySaveConditions() {
+        //we wants that
+        try {
+            // 3) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDataSource( dataSourceDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
+        }
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Same comment about swallowing the exception.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+        verify( updateDataSourceEvent, times( 1 ) ).fire( new UpdateDataSourceEvent( originalDataSourceDef, dataSourceDef, project, SESSION_ID, IDENTITY ) );
+    }
+
+    @Override
+    protected void verifyDeleteConditions() {
+        //we wants that
+        try {
+            // 2) the definition was un-deployed.
+            verify( runtimeManager, times( 1 ) ).unDeployDataSource( dataSourceDeploymentInfo, UnDeploymentOptions.forcedUnDeployment() );
+        } catch ( Exception e ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DataSourceDefEditorServiceTest.java]: Same comment about swallowing the exception.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DefEditorServiceBaseTest.java [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+        //current file
+        String content = getExpectedDefString();
+        when( path.toURI() ).thenReturn( FILE_URI );
+        org.uberfire.java.nio.file.Path nioPath = Paths.convert( path );
+        when( ioService.readAllString( nioPath ) ).thenReturn( content );
+        when( ioService.exists( nioPath ) ).thenReturn( true );
+
+        when ( projectService.resolveProject( path ) ).thenReturn( project );
+
+        try {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DefEditorServiceBaseTest.java]: Do you need a ```try {...} catch( e ) {...}``` block for ```when(..)```?

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DriverDefEditorServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+        return DriverDefSerializer.serialize( originalDriverDef );
+    }
+
+    @Override
+    protected void verifyCreateConditions( boolean global ) {
+        //we wants that:
+        try {
+            // 2) the definition was deployed.
+            verify( runtimeManager, times( 1 ) ).deployDriver( driverDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DriverDefEditorServiceTest.java]: Same comment about swallowing the exception.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DriverDefEditorServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+        verify( newDriverEvent, times( 1 ) ).fire( expectedEvent );
+    }
+
+    @Override
+    protected void verifySaveConditions() {
+        //we wants that
+        try {
+            // 3) the definition was deployed
+            verify( runtimeManager, times( 1 ) ).deployDriver( driverDef, DeploymentOptions.create() );
+        } catch ( Exception e ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DriverDefEditorServiceTest.java]: Same comment about swallowing the exception.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DriverDefEditorServiceTest.java [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+        verify( updateDriverEvent, times( 1 ) ).fire( new UpdateDriverEvent( originalDriverDef, driverDef, project, SESSION_ID, IDENTITY ) );
+    }
+
+    @Override
+    protected void verifyDeleteConditions() {
+        //we wants that
+        try {
+            // 2) the definition was un-deployed.
+            verify( runtimeManager, times( 1 ) ).unDeployDriver( driverDeploymentInfo, UnDeploymentOptions.forcedUnDeployment() );
+        } catch ( Exception e ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/java/org/kie/workbench/common/screens/datasource/management/service/DriverDefEditorServiceTest.java]: Same comment about swallowing the exception.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/resources/META-INF/services/org.uberfire.java.nio.file.spi.FileSystemProvider [hunk 14]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright 2014 JBoss, by Red Hat, Inc
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/test/resources/META-INF/services/org.uberfire.java.nio.file.spi.FileSystemProvider]: Old Skool? 2014? ;-)

**Comment by manstis** [comment]
@wmedvede Nice refactoring!

A few (repeating) comments in some of the test classes; but looks sweet overall.

**Comment by wmedvede** [comment]
@manstis @csadilek  The suggestions were applied.

**Comment by csadilek** [comment]
+1

**Pull request merged**